mllib/src/test/scala/org/apache/spark/ml/tree/impl/TreeTests.scala

```diff
@@ -18,37 +18,46 @@
 package org.apache.spark.ml.tree.impl
 
 import scala.collection.JavaConverters._
+import scala.util.Random
 
 import org.apache.spark.{SparkContext, SparkFunSuite}
 import org.apache.spark.api.java.JavaRDD
 import org.apache.spark.ml.attribute.{AttributeGroup, NominalAttribute, NumericAttribute}
-import org.apache.spark.ml.feature.LabeledPoint
+import org.apache.spark.ml.feature.{Instance, LabeledPoint}
 import org.apache.spark.ml.linalg.Vectors
 import org.apache.spark.ml.tree._
+import org.apache.spark.mllib.util.TestingUtils._
 import org.apache.spark.rdd.RDD
 import org.apache.spark.sql.{DataFrame, SparkSession}
 
 private[ml] object TreeTests extends SparkFunSuite {
 
   /**
    * Convert the given data to a DataFrame, and set the features and label metadata.
+   *
    * @param data  Dataset.  Categorical features and labels must already have 0-based indices.
    *              This must be non-empty.
    * @param categoricalFeatures  Map: categorical feature index to number of distinct values
    * @param numClasses  Number of classes label can take.  If 0, mark as continuous.
    * @return DataFrame with metadata
    */
   def setMetadata(
-      data: RDD[LabeledPoint],
+      data: RDD[_],
```

Comment:
Being test code, this isn't a big deal, but this could also be overloaded with two versions that accept RDDs of different types, and then call to a common private function to proceed from there. I don't feel strongly about it, just dislike seeing `[_]` in a method signature

Reply:
I tried to create two methods with different RDDs, one for LabeledPoint and the other for Instance, but I got an error complaining about duplicate methods (probably due to type erasure).  Maybe I misunderstood, is there a way to make it work?

Reply:
Ah right they'd have to have different names. OK, leave it as is